### PR TITLE
Added Playlist and watch later

### PR DESCRIPTION
**Pull request by Rohanmond**

### CODE REVIEW

File: src/pages/Playlist_Page/components/PlayListCard/PlayListCard.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      <div
+        className='playlist-card-image-container'
+        onClick={() => navigate(`/playlist/${_id}`)}
+      >
+        {videos.length > 0 && (
+          <img
+            className='img-responsive playlist-card-image'
+            src={
+              videos.length > 0
+                ? `https://i.ytimg.com/vi/${videos[0]._id}/maxresdefault.jpg`
```
Comment (varnita619) [inline src/pages/Playlist_Page/components/PlayListCard/PlayListCard.jsx:35]: good use of ternary operator

**Review by varnita619 — APPROVED** [review]
Code is clean with proper naming convension and folder structure. Ready to merge

File: src/contexts/data-context.js [hunk 2]
```diff
@@ -66,6 +87,7 @@ export const DataProvider = ({ children }) => {
         encodedToken: token,
       });
       if (historyRes.status === 200 || historyRes.status === 201) {
+        console.log(historyRes);
```
Comment (Shah-Arsalan) [inline src/contexts/data-context.js]: remove console logs if not needed

File: src/contexts/data-context.js [hunk 3]
```diff
@@ -108,6 +130,142 @@ export const DataProvider = ({ children }) => {
...
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const PostWatchLater = async ({ video }) => {
+    try {
+      if (state.laters.some((el) => el._id === video._id)) {
+        console.log('already added in your watch later');
```
Comment (Shah-Arsalan) [inline src/contexts/data-context.js:279]: here too

File: src/pages/Playlist_Page/components/PlayListCard/PlayListCard.css [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+.playlist-card-number-wrapper {
+  height: 100%;
+  width: 40%;
+  position: absolute;
+  top: 0;
+  right: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background-color: rgba(0, 0, 0, 0.5);
```
Comment (Shah-Arsalan) [inline src/pages/Playlist_Page/components/PlayListCard/PlayListCard.css:47]: use color variables

File: src/pages/Playlist_Page/components/PlayListModal/PlayListModal.css [hunk 5]
```diff
@@ -0,0 +1,64 @@
+.modal-wrapper {
+  width: 100vw;
+  height: 100vh;
+  background-color: rgba(0, 0, 0, 0.3);
```
Comment (Shah-Arsalan) [inline src/pages/Playlist_Page/components/PlayListModal/PlayListModal.css]: here too

File: src/pages/Playlist_Page/components/PlayListModal/PlayListModal.jsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  return (
+    <>
+      {
+        <div className='modal-wrapper'>
+          <div className='playlist-modal' onClick={(e) => e.stopPropagation()}>
+            <div className='playlist-header'>
+              <p>Save to...</p>
+              <span
+                onClick={(e) => {
+                  // e.stopPropagation();
```
Comment (Shah-Arsalan) [inline src/pages/Playlist_Page/components/PlayListModal/PlayListModal.jsx]: remove comments

**Review by Shah-Arsalan — APPROVED** [review]
Code looks modular . Very well written code . You can go ahead with the merge

File: src/App.js [hunk 7]
```diff
@@ -33,7 +41,22 @@ function App() {
               </PrivateRoute>
             }
           />
-          <Route path='playlist_page' element={<PlayList />} />
+          <Route
+            path='playlist_page'
+            element={
+              <PrivateRoute>
```
Comment (vinc3nati) [inline src/App.js:54]: Nice use of protected routes!

**Review by vinc3nati — APPROVED** [review]

**Pull request merged**